60625 site icon UI

kebbet opened the pull request:
First run of updating the UI.

https://core.trac.wordpress.org/ticket/60625

Before
![60625-before](https://github.com/WordPress/wordpress-develop/assets/11491369/8b384eb4-439c-4bba-9f3a-90cfd134a759)

After

![60625-after](https://github.com/WordPress/wordpress-develop/assets/11491369/981b357d-994d-4144-8e67-bf2575e11231)

kebbet committed:
New design of favicon previews.

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

kebbet committed:
Fixes for RTL

mukeshpanchal27 commented on src/wp-admin/options-general.php:
```diff
@@ -155,16 +155,27 @@
-				<img id="browser-icon-preview" src="<?php site_icon_url(); ?>" alt="<?php echo esc_attr( $browser_icon_alt_value ); ?>">
+	<style>
+	:root{
+		--site-icon-url: url( '<?php site_icon_url(); ?>' );
```
`site_icon_url()` called multiple time can we store it value in variable and reuse it?

kebbet replied:
Great observation, solved in [cf5e3d6](https://github.com/WordPress/wordpress-develop/pull/6871/commits/cf5e3d6cdbb7b4b02270cc46aa12276cea7b73ea)

kebbet committed:
Store value of site_icon_url  in variable.

kebbet committed:
Adjusts crop preview to work in both LTR and RTL.

kebbet committed:
Use get_site_icon_url

kebbet committed:
Use get_site_icon_url, 4realz.

kebbet commented:
@mukeshpanchal27, whats your opinion, should we remove both these assets: `browser-rtl.png` and `browser.png`?

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

kebbet committed:
Remove browser preview images (since they are not referenced anymore)

kebbet committed:
Add removed files to $_old_files

afercia commented on src/wp-admin/css/site-icon.css:
```diff
@@ -2,55 +2,154 @@
-	width: 72px;
-	white-space: nowrap;
+.site-icon-preview.hidden {
+	visibility: none;
```
Not sure `visibility` has a `none` value. Anyways, this is probably unnecessary because of `display: none;` two lines below?

kebbet replied:
Removed line 39 and 40.

afercia commented on src/wp-includes/media-template.php:
```diff
@@ -1542,21 +1542,31 @@ function wp_print_media_templates() {
+			}
+		</style>
+		<h2><?php _ex( 'Site Icon Preview', 'noun' ); ?></h2>
+		<p aria-hidden="true"><?php _e( 'As a browser icon and an app icon.' ); ?></p>
```
I wouldn't hide this paragraph from assistive technologies. Although the two following images do have meaningful alt text, letting this text be read by screen readers doesn't harm.

kebbet replied:
Removed aria-hidden.

afercia commented:
I'd like to suggest to take the opportunity of this PR to chamge a couple translatable strings, specifically the ones that use th the recommended size with this pattern: `512 &times; 512`.
The problem with `&times;` is that, while visually looks like an `x`, it is ready by screen readers as `times`.
In the past some similar ases in core have been updated ot use `by` so that it will bread `512 by 512 pixels`. I would also consider to remove the placeholder and make the string fully translatable. [Here](https://github.com/WordPress/wordpress-develop/blob/5effffccd005ca3dae6631ee73b1096b8f54a861/src/wp-admin/options-general.php#L202) and [here](https://github.com/WordPress/wordpress-develop/blob/5effffccd005ca3dae6631ee73b1096b8f54a861/src/wp-includes/class-wp-customize-manager.php#L5207-L5208).

kebbet committed:
Removes unneeded css rules.

kebbet committed:
Do not hide paragraph for screen readers.

afercia commented:
A few minor things:

In the media modal dialog preview;

![mieda modal preview](https://github.com/WordPress/wordpress-develop/assets/1682452/34e120ed-14f3-4469-8c3a-d8409aee8fba)

The preview shows the app icon first and the browser icon last. However, the description `As a browser icon and an app icon.` mentions the two icons in a reversed order, which may be a little confusing. I'd suggest to change the description to use the same order of the visual preview order: `As an app icon and a browser icon.`


The customizer media control shows the Remove button first and the Change image button last:

![customizer](https://github.com/WordPress/wordpress-develop/assets/1682452/ab15f614-8410-4714-b913-8a0fef49dc4c)

While in the Settings page the order is different:

![settings page](https://github.com/WordPress/wordpress-develop/assets/1682452/d929a7cb-c17d-4c29-830d-e27ff58daadf)

If possible, I would use the same order everywhere. Not sure if that would impact all the customizer media controsl though.
It's a minor thing though also because the Site icon UI is typically used once or very few times.

Noting that also the description placement in the customizer is different but I'm not sure that can be changed.

kebbet committed:
Revert unintended changes

afercia reviewed (commented):
From an accessibility perspective everything looks good to me, thanks!
I quickly tested the functionality and didn't notice any issue

I left some minor notes. I'll leave a more complete code review to someone else more familiar than me with the customizer.

kebbet committed:
Reflect wording with the graphics order below.

kebbet commented:
> A few minor things:
> 
> In the media modal dialog preview;
> 
> [IMAGE]
> The preview shows the app icon first and the browser icon last. However, the description `As a browser icon and an app icon.` mentions the two icons in a reversed order, which may be a little confusing. I'd suggest to change the description to use the same order of the visual preview order: `As an app icon and a browser icon.`

Great feedback, changed that.

> 
> The customizer media control shows the Remove button first and the Change image button last:
> 
> [IMAGE] 
> While in the Settings page the order is different:
> 
>[IMAGE]
> If possible, I would use the same order everywhere. Not sure if that would impact all the customizer media controsl though. It's a minor thing though also because the Site icon UI is typically used once or very few times.

The site logo controller in the customizer share the same pattern as site icon controller. So i think thats a separate ticket.

> 
> Noting that also the description placement in the customizer is different but I'm not sure that can be changed.

That can be changed, which the screenshot shows, and I removed the long italic style, plus changed to 512 _by_ 512 in the description.

![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/917b9a82-21e0-4bc9-92c1-c1a91bc56b81)

kebbet committed:
Customizer, updates description string, style and position.

kebbet committed:
Make the sizes translateable.

afercia commented:
>  I removed the long italic style

Good catch, thank you.

mukeshpanchal27 commented on .env:
```diff
@@ -51,7 +51,7 @@ LOCAL_DB_TYPE=mysql
 # When using `mariadb`, see https://hub.docker.com/_/mariadb for valid versions.
 ##
-LOCAL_DB_VERSION=8.0
+LOCAL_DB_VERSION=8
```
Unwanted changes 🤔

kebbet replied:
Thanks for spotting. misssed it.

kebbet committed:
revert unwanted xhange

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -2,55 +2,152 @@
+	flex: 1 0 0;
+	z-index: 1;
+	border-top-left-radius: 10px;
+	border-top: 1px solid rgba(255, 255, 255, 0.20);
+	border-left: 1px solid rgba(255, 255, 255, 0.20);
```
0.2 instead of 0.20

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -2,55 +2,152 @@
+.site-icon-preview-browser .browser-icon-preview {
+	width: 24px;
+	height: 24px;
+	box-shadow: 0 0 20px 0 rgba(0,0,0,0.1);
```
Spacing: 0, 0, 0, 0.1

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -2,55 +2,152 @@
+	border-top: 1px solid rgba(255, 255, 255, 0.20);
+	border-left: 1px solid rgba(255, 255, 255, 0.20);
+	background: linear-gradient(180deg, #dcdcdc 0%, #bdbdbd 100%);
+	box-shadow: 0px 10px 22px 0px rgba(0, 0, 0, 0.20);
```
0.2, not 0.20

kebbet commented on src/wp-admin/options-general.php:
```diff
@@ -121,6 +121,7 @@
 
 	// Handle alt text for site icon on page load.
```
Update comment?

kebbet committed:
Fixes some css coding standard issues.

kebbet committed:
Update docblock and remove one call to get_site_icon_url()

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

kebbet committed:
Remove changes due to sync issues.

kebbet committed:
Remove changes due to sync issues.

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

kebbet committed:
Re-add removed files to old_files array.

peterwilsoncc commented on src/wp-admin/options-general.php:
```diff
@@ -155,16 +158,27 @@
-				<img id="browser-icon-preview" src="<?php site_icon_url(); ?>" alt="<?php echo esc_attr( $browser_icon_alt_value ); ?>">
+	<style>
+	:root{
+		--site-icon-url: url( '<?php echo esc_url( $site_icon_url ); ?>' );
```
For URLs that include query strings, this will encode the `&` characters to `&#038;`. eg `http://wp-dev.local/image.png?a=b&c=d` becomes `http://wp-dev.local/image.png?a=b&#038;c=d`.

I'm not sure if that's expected within `style` tags so it would be good to test against the spec. If it's not expected then use `sanitize_url()` instead.

This may not be an issue in core but for sites using services similar to Jetpack's Photon it could be.

kebbet replied:
Good question! When adding `?a=b&#038;c=d` to the image string, everyting seems to work fine in Firefox, Safari and Chrome. So maybe keep it?

peterwilsoncc commented on src/wp-admin/options-general.php:
```diff
@@ -198,8 +212,12 @@ class="<?php echo esc_attr( $classes_for_button ); ?>"
-			printf( __( 'The Site Icon is what you see in browser tabs, bookmark bars, and within the WordPress mobile apps. It should be square and at least %s pixels.' ), '<code>512 &times; 512</code>' );
+			/* translators: 1: opening  <code> tag. 2: closing </code> tag. */
+			printf(
+				__( 'The Site Icon is what you see in browser tabs, bookmark bars, and within the WordPress mobile apps. It should be square and at least %1$s512 by 512%2$s pixels.' ),
```
With the change of phrase, are the code tags still required? I have no idea...

![Screenshot 2024-07-31 at 12 51 39 PM](https://github.com/user-attachments/assets/489858c4-b34b-490a-a4cc-62daf39937ae)

kebbet replied:
It makes the values stand out a bit more on the canvas. I like it. With removed code tags, the values get a bit more anonymous...

kebbet replied:
<img width="265" alt="bild" src="https://github.com/user-attachments/assets/a284191a-f79d-42a1-894b-2c570c35ab49">

peterwilsoncc commented on src/wp-admin/options-general.php:
```diff
@@ -155,16 +158,27 @@
+	}
+	</style>
+
+	<div id="site-icon-preview" class="site-icon-preview settings <?php echo esc_attr( $classes_for_wrapper ); ?>">
```
This will need a right to left version

![Screenshot 2024-07-31 at 12 56 43 PM](https://github.com/user-attachments/assets/7fd10795-b4cb-4fe4-b7fd-0408c882f549)

kebbet replied:
It has a version, from what I can create with the watch script running.

kebbet replied:
<img width="846" alt="bild" src="https://github.com/user-attachments/assets/a744aade-b2a0-4076-864c-cfabaabc9ba4">

peterwilsoncc replied:
> It has a version, from what I can create with the watch script running.

Sorry, it appears I was hitting a caching bug in style concatenation. 

To represent RTL browsers, should the tab be reversed and the window UI appear to the right?

![Screenshot 2024-08-01 at 8 18 21 AM](https://github.com/user-attachments/assets/ff916dbc-3dd5-4c9f-86bf-553f0668ce3a)


Please excuse the dodgy mock up.

kebbet replied:
That makes sence! I did some searching for screenshots of real RTL environments, and the layout you mocked up looks more like it! Attaching a screenshot of my testings with (not yet commited) updated CSS.

![bild](https://github.com/user-attachments/assets/26c0b95b-a0d3-4dd3-8070-79900bf96322)

kebbet replied:
Marking as resolved with [329823b](https://github.com/WordPress/wordpress-develop/pull/6871/commits/329823b7a63cecc0c027444132d57c995d2cb1ad)

peterwilsoncc reviewed (commented):
I've added a few notes, mainly questions about escaping rules within style tags.

The new design looks really nice, thanks.

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

kebbet committed:
CSS refinements.

kebbet commented:
Thanks for taking the time for testing and reviewing @peterwilsoncc

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

kebbet committed:
Fix layout of browser for RTL environments.

kebbet commented:
This is a screenshot with the current state of the PR.

![60625-after2](https://github.com/user-attachments/assets/165e0246-84d0-440c-b7f6-df357982cfdf)

peterwilsoncc commented on src/wp-admin/includes/update-core.php:
```diff
@@ -770,6 +770,9 @@
 	'wp-includes/blocks/block/editor.min.css',
 	'wp-includes/blocks/block/editor-rtl.css',
 	'wp-includes/blocks/block/editor-rtl.min.css',
+	// 6.7
+	'wp-admin/images/browser-rtl.png',
+	'wp-admin/images/browser.png',
```
The list of old files is updated at the end of the release cycle to ensure sites running trunk don't lose files in commits that are subsequently reverted (an edge case, I know).

```suggestion
```

peterwilsoncc reviewed (approved):
This looks great, thanks for your work on this.

I've added one final note which I'll commit in a moment.

peterwilsoncc committed:
Remove old files block for an update at the end of the release cycle.

kebbet committed:
Tweak "Choose a Site Icon" button look. Fix safari color issue.

kebbet committed:
Target html-tag with style, not :root.

kebbet committed:
Sets colors as CSS vars. Adds dark mode style.

kebbet commented:
While looking at @jameskoster original comment i notices the suggestion to make the colors to CSS-vars. Latest commit implements that. 

The commit also takes advantage of the CSS vars, and implements a dark mode style. See attached image.

![bild](https://github.com/user-attachments/assets/9f9f7ad0-2413-420c-96e8-e4fddd5d49bf)

kebbet committed:
Use :root for crop preview to work.

aaronjorbin commented on src/wp-admin/options-general.php:
```diff
@@ -190,16 +204,20 @@ class="<?php echo esc_attr( $classes_for_button ); ?>"
-			printf( __( 'The Site Icon is what you see in browser tabs, bookmark bars, and within the WordPress mobile apps. It should be square and at least %s pixels.' ), '<code>512 &times; 512</code>' );
+			/* translators: 1: opening  <code> tag. 2: closing </code> tag. */
+			printf(
+				__( 'The Site Icon is what you see in browser tabs, bookmark bars, and within the WordPress mobile apps. It should be square and at least %1$s512 by 512%2$s pixels.' ),
```
the `512  &times; 512` was intentionally not translated since those values aren't changeable.  I think we should keep them as passed values.

kebbet replied:
Thank you for the testing, I've addressed this concern in my latest commit. => [68e0a5b](https://github.com/WordPress/wordpress-develop/pull/6871/commits/68e0a5b7e237768ef05ddec98df7fc4528da201a)

peterwilsoncc replied:
> the `512 &times; 512` was intentionally not translated since those values aren't changeable. I think we should keep them as passed values.

According to a [random page I found on the internet](https://www.omniglot.com/language/numerals.htm) (so it must be true 🙂) there are different numerals in use for non-latin languages. It would be good to get some input from some i18n experts to find out if the string ought to be translatable. cc @swissspidy @ocean90

kebbet replied:
Maybe a case for the classic `number_format_i18n` which have been around since 2.3?

swissspidy replied:
There's precedence in core for not translating this, so let's keep the existing text at is.

But **please** do not use placeholders for HTML tags! Undo that change.

kebbet replied:
Ok @swissspidy. New changes committed.

aaronjorbin reviewed (commented):
Tested this, it's looking good. Thanks! The only thought I have is that it would be best to keep the 512 as unchangeable in the translation since it's not a value that can be changed in the code.

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

kebbet committed:
Don't make site icon size recommendation changeable in translation.

kebbet committed:
Include html tag in translatable string

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

swissspidy commented on src/wp-admin/options-general.php:
```diff
@@ -124,9 +124,12 @@
 	$app_icon_alt_value     = '';
 	$browser_icon_alt_value = '';
 
+	// Store icon URL as variables for use in multiple places.
```
Nit: this is not really a useful comment. The code is self-explanatory as-is.

kebbet replied:
Ok!

swissspidy commented on src/wp-admin/options-general.php:
```diff
@@ -155,20 +158,31 @@
+				<svg role="img" aria-hidden="true" fill="none" xmlns="http://www.w3.org/2000/svg" class="browser-buttons"><path fill-rule="evenodd" clip-rule="evenodd" d="M0 20a6 6 0 1 1 12 0 6 6 0 0 1-12 0Zm18 0a6 6 0 1 1 12 0 6 6 0 0 1-12 0Zm24-6a6 6 0 1 0 0 12 6 6 0 0 0 0-12Z" /></svg>
+				<div class="site-icon-preview-tab">
+					<img id="browser-icon-preview" src="<?php echo esc_url( $site_icon_url ); ?>" class="browser-icon-preview" alt="<?php echo esc_attr( $browser_icon_alt_value ); ?>" />
+					<div class="site-icon-preview-site-title" id="site-icon-preview-site-title" aria-hidden="true"><?php echo get_bloginfo( 'name' ); ?></div>
```
Why change from `bloginfo( 'name' )` to `echo get_bloginfo( 'name' )`? They're slightly different

kebbet replied:
No real argument for it, I will add back `bloginfo( 'name' )`.

kebbet committed:
Changes requested.

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

kebbet committed:
Merge branch 'WordPress:trunk' into 60625-site-icon-ui

peterwilsoncc committed:
Merge branch 'trunk' into 60625-site-icon-ui

peterwilsoncc commented:
Merged [r58927](https://core.trac.wordpress.org/changeset/58927) / https://github.com/WordPress/wordpress-develop/commit/3da83600189bc98661110b32dc488e753c8d622d

Thanks for your patience during the review process.
